capstone_project/src/main/webapp/scripts/mapsScriptTemplate.js

```diff
@@ -174,13 +174,63 @@ function createNewsFeed(articles) {
   return newsFeed;
 }
 
+/** Converts post objects into links to be featured under the "Posts" tab.
+ * @param {Array.<Object>} schoolPosts - A list of posts pertaining to a school.
+ * @param {string} schoolName - The name of a school.
+ * @return {string} - The HTML code for providing either post links or an option
+ * to make the first post for a school.
+ */
+function createLinks(schoolPosts, schoolName) {
+  // If there are no existing posts for a school, provide the user with a link
+  // to the page where they can create one.
+  if (schoolPosts === undefined || schoolPosts.length === 0) {
+    return `<h5>
+                There are no posts for ${schoolName} yet. Click
+                <a href="../pages/comments.jsp">here</a>
+                to be the first to make one.
+            </h5>`;
+  }
+
+  // Generate an unordered list of links to individual posts.
+  let postFeed = `<h5>
+                    <b>Check out posts from students at ${schoolName}:</b>
+                </h5>`;
+
+  postFeed += '<ul>';
+  for (const post of schoolPosts) {
+    postID = post.key.id;
+    postFeed += `<li>
+                   <a href="../pages/post_displayer.html?post-id=${postID}">
+                     ${post.propertyMap.title}
+                   </a>
+                </li>`;
+  }
+  postFeed += '</ul>';
+  return postFeed;
+}
+
+/** Fetches a school's posts and creates the "Posts" tab portion of a content
+ * string for a marker.
+ * @param {string} name - The name of a school.
+ * @return {string} - A postFeed string containing the HTML code for
+ * the "Posts" tab of an infowindow.
+ */
+async function createPostsFeed(name) {
+  let schoolPosts;
+  await fetch(`/fetch-school-posts?school-name=${encodeURIComponent(name)}`)
+      .then((res) => res.json())
+      .then((data) => schoolPosts = createLinks(data, name));
```

Comment:
I think, not entirely sure, but I think you can rewrite this to:
```suggestion
  let schoolPosts = await fetch(`/fetch-school-posts?school-name=${encodeURIComponent(name)}`)
      .then((res) => res.json())
      .then((data) => createLinks(data, name));
```
since createLinks function returns a string

Reply:
@Leoism yup, this worked!